guide_algorithm_gaussian_process.cpp

```diff
@@ -952,12 +954,21 @@ double GuideAlgorithmGaussianProcess::result(double input)
         return deduceResult();
     }
 
-    // the third parameter of result() is a floating-point in seconds, while RequestedExposureDuration() returns milliseconds
-    const Star& star = pFrame->pGuider->PrimaryStar();
-    double control_signal = GPG->result(input, star.SNR, (double) pFrame->RequestedExposureDuration() / 1000.0);
-
-    Debug.Write(wxString::Format("PPEC: input: %.2f, control: %.2f, exposure: %d\n",
-        input, control_signal, pFrame->RequestedExposureDuration()));
+    try
```

Comment:
I don't think we want to take this change for a couple reasons:
 1. the catch block code is unreachable because none of the enclosed functions will ever throw an instance of `wxString`.
 2. it introduces a precedent implying that it is OK to throw an exception out of a method and that callers should be expected to handle those exceptions.

The code base currently uses exceptions only for control flow _within_ a function or method, and we do not allow exceptions to propagate. For error propagation we use good old fashioned C-style error return codes to make the code much easier to reason about--we generally do not need to be concerned with C++ exception handling.

Suggestion: it looks like it may be possible for some of the eigen code to throw a `std::runtime_exception`, so if we want to handle that we could add a catch block (or blocks) inside `GaussianProcessGuider::result`.

Let's address this part as a separate PR since it is unrelated to the rest of the changes in this pr.

Reply:
Agreed, it was a bad idea.  I've reverted the change in the repo or at least I think I have.